client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -343,4 +344,9 @@
     * Executes a remote script passing a set of named parameters
     */
    <T> T execute(String scriptName, Map<String, ?> params);
+
+   /**
+    * Returns segment ownership for this cache
+    */
+   Map<SocketAddress, Set<Integer>> getSegmentsByServer();
```

Comment:
It is definitely not a user-oriented API: however, once it ends up in the public API we have to keep it stable